File: android/src/main/java/com/azifacemobile/models/ProcessorResult.java
package com.azifacemobile.models;

import com.azifacemobile.models.abstracts.CommonParams;
import com.facebook.react.bridge.Arguments;
import com.facebook.react.bridge.WritableMap;

import org.json.JSONObject;

public class ProcessorResult extends CommonParams {
  private final WritableMap object;

  public ProcessorResult(JSONObject target) {
    super(target);

    this.object = Arguments.createMap();

    this.parseJSON();
  }

  private void parseJSON() {
    Integer ageV2GroupEnumInt = this.getParam("ageV2GroupEnumInt", Integer.class);
    Boolean livenessProven = this.getParam("livenessProven", Boolean.class);
    String auditTrailImage = this.getParam("auditTrailImage", String.class);
    Integer matchLevel = this.getParam("matchLevel", Integer.class);

    this.object.putInt("ageV2GroupEnumInt", this.defaultIfNull(ageV2GroupEnumInt, -1));
    this.object.putBoolean("livenessProven", Boolean.TRUE.equals(livenessProven));
    this.object.putString("auditTrailImage", auditTrailImage);
    this.object.putInt("matchLevel", this.defaultIfNull(matchLevel, -1));
  }

  public WritableMap getMap() {
    return this.object.copy();
  }
}

File: ios/Models/ProcessorResult.swift
import Foundation

public class ProcessorResult: JSONParams {
  private var object: [AnyHashable: Any?]

  override init(target: [AnyHashable: Any]?) {
    self.object = [:]

    super.init(target: target)

    self.parseJSON()
  }

  private func parseJSON() {
    let ageV2GroupEnumInt = self.getParam("ageV2GroupEnumInt", type: Int.self)
    let livenessProven = self.getParam("livenessProven", type: Bool.self)
    let auditTrailImage = self.getParam("auditTrailImage", type: String.self)
    let matchLevel = self.getParam("matchLevel", type: Int.self)

    self.object["ageV2GroupEnumInt"] = self.defaultIfNull(ageV2GroupEnumInt, defaultValue: -1)
    self.object["livenessProven"] = livenessProven == true
    self.object["auditTrailImage"] = auditTrailImage
    self.object["matchLevel"] = self.defaultIfNull(matchLevel, defaultValue: -1)
  }

  func getDict() -> NSDictionary {
    return NSDictionary(dictionary: self.object as [AnyHashable: Any])
  }
}
